bee-ledger/src/workers/pruning/batch.rs

```diff
@@ -197,18 +208,20 @@ pub fn batch_prunable_confirmed_data<S: StorageBackend>(
     metrics.prunable_messages = visited.len();
     metrics.new_seps = new_seps.len();
 
-    Ok((new_seps, metrics))
+    Ok((new_seps, metrics, byte_length))
 }
 
-pub fn batch_prunable_unconfirmed_data<S: StorageBackend>(
+pub(crate) fn batch_prunable_unconfirmed_data<S: StorageBackend>(
     storage: &S,
     batch: &mut S::Batch,
     prune_index: MilestoneIndex,
-) -> Result<UnconfirmedDataPruningMetrics, Error> {
+    by_size: bool,
+) -> Result<(ByteLength, UnconfirmedDataPruningMetrics), PruningError> {
+    let mut byte_length = 0usize;
     let mut metrics = UnconfirmedDataPruningMetrics::default();
 
     let unconf_msgs = match Fetch::<MilestoneIndex, Vec<UnreferencedMessage>>::fetch(storage, &prune_index)
-        .map_err(|e| Error::Storage(Box::new(e)))?
+        .map_err(|e| PruningError::Storage(Box::new(e)))?
```

Comment:
maybe there should be a dedicated method in `PruningError` to do this

Reply:
I am not quite sure what you mean? Can you elaborate? Do you mean the boxing?

Reply:
yeah I'd do something like

```rust
impl PruningError {
    fn storage(e: impl Error) -> Self {
        Self::Storage(Box::new(e))
    }
}
```
so you can do `.map_err(PruningError::storage)` everywhere instead.

Reply:
Will do!